pkg/collector/process.go

```diff
@@ -0,0 +1,283 @@
+package collector
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"github.com/vmware/go-ipfix/pkg/entities"
+	"github.com/vmware/go-ipfix/pkg/registry"
+	"io"
+	"k8s.io/klog"
+	"net"
+	"sync"
+	"time"
+)
+
+type collectingProcess struct {
+	// for each obsDomainID, there is a map of templates
+	templatesMap map[uint32]map[uint16][]*TemplateField
+	// templatesLock allows multiple readers or one writer at the same time
+	templatesLock *sync.RWMutex
+	// template lifetime
+	templateTTL uint32
+	// registries for decoding Information Element
+	ianaRegistry   registry.Registry
+	antreaRegistry registry.Registry
+	// server information
+	address net.Addr
+	// maximum buffer size to read the record
+	maxBufferSize uint16
+	// chanel to receive stop information
+	stopChan chan bool
+	// packet list
+	packets []*Packet
+}
+
+type UDPCollectingProcess struct {
+	*collectingProcess
+	// list of udp workers to process message
+	workerList []*Worker
+	// udp worker pool to receive and process message
+	workerPool chan chan *bytes.Buffer
+	// number of workers
+	workerNum int
+}
+
+type TCPCollectingProcess struct {
+	*collectingProcess
+}
+
+// data struct of processed message
+type Packet struct {
+	Version      uint16
+	BufferLength uint16
+	SeqNumber    uint32
+	ObsDomainID  uint32
+	ExportTime   uint32
+	FlowSets     interface{}
```

Comment:
why not just "Sets"?

Reply:
I was trying to differentiating it from Set. Since we will use Set as the interface, using Sets makes more sense.